rpcclient/infrastructure.go

```diff
@@ -756,14 +759,31 @@ func (c *Client) handleSendPostMessage(jReq *jsonRequest) {
 	if !c.config.DisableTLS {
 		protocol = "https"
 	}
-	url := protocol + "://" + c.config.Host
 
 	var (
 		err, lastErr error
 		backoff      time.Duration
 		httpResponse *http.Response
 	)
 
+	parsedAddr, err := ParseAddressString(c.config.Host, "")
+	if err != nil {
+		jReq.responseChan <- &Response{
+			err: fmt.Errorf("failed to parse address %v", err),
+		}
+		return
+	}
+
+	var url string
+	switch parsedAddr.Network(){
+	case "unix", "unixpacket":
+		// Using a placeholder URL because a non-empty URL is required
```

Comment:
nit: this sounds like they are two separate sentences, so should add punctuation/capitalization accordingly.